index.js

```diff
@@ -1,4 +1,23 @@
-(function() {
+(function(factory) {
+
+  if (typeof exports !== 'undefined') {
+    var _factory = factory();
+    module.exports = _factory;
+    module.exports.Factory = _factory.Factory;
+    module.exports.Adapter = _factory.Adapter;
+    module.exports.ObjectAdapter = require('./lib/object-adapter');
+  }
+  else if (typeof define === 'function' && define.amd) {
+    define(['factory-girl-object-adapter'], function(ObjectAdapter) {
```

Comment:
I don't know if I understood right, but it's almost the same code that you had to export FactoryGirl.
I made another commit to remove some assignments that was unnecessary because it was done on instantiation at the end of the file.